### PR TITLE
Allowing services to be disabled from enumeration

### PR DESCRIPTION
**Pull request by BSick7**
## Problem

This tool is amazing, but running an enumeration requires either all services or none.
There are some services I don't want to run.

## Solution

This PR introduces an ability to disable a service by name.

### CODE REVIEW

File: amass/config.go [hunk 1]
```diff
@@ -162,3 +165,21 @@ func (c *AmassConfig) GetAPIKey(source string) *APIKey {
...
+				include = false
+				break
+			}
+		}
+		if include {
+			enabled = append(enabled, service)
+		}
+	}
+	return enabled
+}
```
Comment (kalbasit) [inline amass/config.go]: Please add a test for this asserting both valid and invalid input.

File: amass/config.go [hunk 2]
```diff
@@ -48,6 +48,9 @@ type AmassConfig struct {
 	// A blacklist of subdomain names that will not be investigated
 	Blacklist []string
 
+	// List of services that should not be run
+	DisabledServices []string
```
Comment (kalbasit) [inline amass/config.go]: @caffix are these config variables accessible from outside the amass package?

Comment (caffix) [reply amass/config.go]: Yes. Each enumeration has a config that is referenced throughout its execution.

I would prefer that the new configuration field be named DisabledDataSources

**Comment by caffix** [comment]
@BSick7 I completely agree with you, and was going to implement this feature before the end of the 2.8 minor version. I like your implementation, and will have the command-line tool utilize it

File: amass/config.go (hunk 1 above)
Comment (BSick7) [reply amass/config.go]: Done.

File: amass/config.go (hunk 2 above)
Comment (BSick7) [reply amass/config.go]: Will do.

**Comment by BSick7** [comment]
@caffix Renamed to `DisabledDataSources` and added a test.

**Comment by caffix** [comment]
This PR has become a bit messy. I'll create a new one

**Pull request closed without merging**